src/modules/News/controller.ts

```diff
@@ -13,6 +17,35 @@ const getLatestNews = async (req: Request, res: Response) => {
   }
 };
 
+const getPersonalizedNews = async (req: Request, res: Response) => {
+  const { categories, limit, page } = req.query;
+  const category = validateCategory(categories as string);
+
+  const pageNumber = Number(page) || 1;
+  const limitNumber = Number(limit) || 10;
+
+  try {
+    const news = await fetchFromNewsAPI('/top-headlines', {
+      category,
+      pageSize: limitNumber,
+      page: pageNumber,
+    });
+
+    const paginatedArticles: PaginatedResults<IArticles> =
+      formatArticlesResponse(news.articles, pageNumber, limitNumber);
+
+    res.status(200).send({
+      articles: paginatedArticles.data,
+      currentPage: paginatedArticles.currentPage,
+      totalPages: paginatedArticles.totalPages,
+    });
+  } catch (error) {
+    res
+      .status(500)
+      .send({ message: 'Error in getting personalized news', error });
+  }
+};
+
```

Comment:
Comment 1: The commit that contains these parts of the code shouldn't be in this PR, as they are related to the other [PR ](https://github.com/StartSteps-Digital-Education-GmbH/News-App/pull/14) you are working on.
When working on a new feature always create the new branches from the `main` branch.
For now, it's fine but we have to merge your personalized news branch first, then this one.